service/src/main/java/bio/terra/tanagra/service/AccessControlService.java

```diff
@@ -27,7 +33,11 @@ public void throwIfUnauthorized(
       Action action,
       ResourceType resourceType,
       @Nullable ResourceId resourceId) {
-    if (!isAuthorized(credential, action, resourceType, resourceId)) {
+    // TODO: Update signature to take underlayName parameter once all routes have been configured to
+    // require an underlay.
```

Comment:
The problem doesn't exist in `PluginService`, it exists in the classes calling `PluginService`. I can update `getPlugin()` to ignore the underlay parameter, but that would pretty much invalidate any tests done against the service.

Reply:
Hmm maybe I don't understand how this will work then. If you hard-code the underlay to empty string, then won't this throw an exception for every endpoint that does an access control check?

My initial read was that you wanted a placeholder value for the underlay name so you can tackle plumbing it through to the controllers in a future PR. What I was trying to say is that I would just do the hard-coding (and the special casing that I think you'll need to handle an invalid underlay name) in `PluginService` rather than here. You can still pass the underlay name for your tests.

e.g. Have 2 overloaded methods there:
  - One takes an underlay name that you call in your tests
  - Another takes no arguments and handles returning the defaults until you've got the underlay name plumbed through everywhere.

Reply:
I don't think a default plugin should be able to be loaded if there is configured version of that plugin available. That means an underlay should be required to get a plugin.

I'll just delete all the calls to the plugin service in `AccessControlService` until the all the endpoints can be updated to require an underlay.